content/tutorials/build-apps/build-a-desktop-wallet-in-javascript.md

```diff
@@ -0,0 +1,1433 @@
+        </table>
+    </fieldset>
+    <!-- Step 4 code additions - end -->
+
+    <dialog id="account-address-dialog">
+        <form method="dialog">
+```
+
+The table here will be filled dynamically with the accounts transactions.
+
+6. Add the following code at the bottom of `view/renderer.js`:
+
+```javascript
+const txTableBodyEl = document.getElementById('tx-table').tBodies[0]
+window.testEl = txTableBodyEl
+
+window.electronAPI.onUpdateTransactionData((_event, transactions) => {
+    for (let transaction of transactions) {
+        txTableBodyEl.insertAdjacentHTML( 'beforeend',
+        "<tr>" +
+            "<td>" + transaction.confirmed + "</td>" +
+            "<td>" + transaction.type + "</td>" +
+            "<td>" + transaction.from + "</td>" +
+            "<td>" + transaction.to + "</td>" +
+            "<td>" + transaction.value + "</td>" +
+            "<td>" + transaction.hash + "</td>" +
+            "</tr>"
+        )
+    }
+})
+```
+
+If you have come this far - congrats. Run the application by executing:
+
+```console
+npm run start
+```
+
+Our application should by now display all the stuff from the last step plus an additional list of transactions associated with the given account.
+
+To run the reference application found in `content/_code-samples/build-a-wallet/desktop-js` for this step, run:
+
+```console
+npm run tx-history
+```
+
+### 5. Saving the Private Keys with a Password
+
+**Full code for this step:**
+[`5_password.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/5_password.js).
+[`library/3_helper.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/library/3_helper.js).
+[`library/4_helper.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/library/4_helper.js).
+[`library/5_helper.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/library/5_helper.js).
+[`view/5_preload.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/5_tx-preload.js).
+[`view/5_password.html`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/5_password.html).
+[`view/5_renderer.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/desktop-js/view/5_tx-renderer.js).
+
+After finishing this step the application should look like this:
+
+![Screenshot: Step 5, use salted password](img/javascript-wallet-5.png)
+
+By now we always query the user for an account address at application startup. We more or less have a monitoring tool for accounts that queries publicly available data. Because we want to have real wallet functionality including sending XRP, we will have to deal with private keys and seeds, which will have to be handled properly.
```

Comment:
nit: The last bit was said already in this paragraph
```suggestion
By now we always query the user for an account address at application startup. We more or less have a monitoring tool for accounts that queries publicly available data. Because we want to have real wallet functionality including sending XRP, we will have to deal with private keys and seeds.
```